aospy/test/test_data_loader.py

```diff
@@ -0,0 +1,392 @@
+        self.assertEqual(result, expected)
+
+    def test_data_name_gfdl_monthly(self):
+        # Test 'ts' and 'inst' data
+        for data_type in ['ts', 'inst']:
+            expected = 'atmos.200601-201112.temp.nc'
+            result = io.data_name_gfdl('temp', 'atmos', data_type,
+                                       'monthly', 2010, 'jja', 2000, 6)
+            self.assertEqual(result, expected)
+
+        # Test 'av' data
+        for intvl_type in ['monthly', 'mon']:
+            expected = 'atmos.2010.jja.nc'
+            result = io.data_name_gfdl('temp', 'atmos', 'av',
+                                       intvl_type, 2010, 'jja', 2000, 1)
+            self.assertEqual(result, expected)
+
+            expected = 'atmos.2006-2011.jja.nc'
+            result = io.data_name_gfdl('temp', 'atmos', 'av',
+                                       intvl_type, 2010, 'jja', 2000, 6)
+            self.assertEqual(result, expected)
+
+        # Test 'av_ts' data
+        expected = 'atmos.2006-2011.01-12.nc'
+        result = io.data_name_gfdl('temp', 'atmos', 'av_ts',
+                                   'monthly', 2010, 'jja', 2000, 6)
+        self.assertEqual(result, expected)
+
+    def test_data_name_gfdl_daily(self):
+        # Test 'ts' and 'inst' data
+        for data_type in ['ts', 'inst']:
+            expected = 'atmos.20060101-20111231.temp.nc'
+            result = io.data_name_gfdl('temp', 'atmos', data_type,
+                                       'daily', 2010, None, 2000, 6)
+            self.assertEqual(result, expected)
+
+        # Test 'av' data; this should raise an error
+        with self.assertRaises(NameError):
+            io.data_name_gfdl('temp', 'atmos', 'av',
+                              'daily', 2010, None, 2000, 6)
+
+        # Test 'av_ts' data
+        expected = 'atmos.2006-2011.01-12.nc'
+        result = io.data_name_gfdl('temp', 'atmos', 'av_ts',
+                                   'daily', 2010, None, 2000, 6)
+        self.assertEqual(result, expected)
+
+    def test_data_name_gfdl_hr(self):
+        # Test 'ts' and 'inst' data
+        for data_type in ['ts', 'inst']:
+            expected = 'atmos.2006010100-2011123123.temp.nc'
+            result = io.data_name_gfdl('temp', 'atmos', data_type,
+                                       '3hr', 2010, None, 2000, 6)
+            self.assertEqual(result, expected)
+
+        # Test 'av' data; this should raise an error
+        with self.assertRaises(NameError):
+            io.data_name_gfdl('temp', 'atmos', 'av',
+                              '3hr', 2010, None, 2000, 6)
+
+        # Test 'av_ts' data
+        expected = 'atmos.2006-2011.01-12.nc'
+        result = io.data_name_gfdl('temp', 'atmos', 'av_ts',
+                                   '3hr', 2010, None, 2000, 6)
+        self.assertEqual(result, expected)
+
+    @unittest.expectedFailure  # This aspect of this function is obselete
```

Comment:
Gotcha.  This works and the test passes by just changing this to `label = intvl.upper()`.  I don't use these file types often but they are very regularly outputted, so let's go that route.